Crash when denying location access twice

## 🐛 Bug Report

First deny returns exception as expected
```PlatformException(PERMISSION_DENIED, Access to location data denied, null)```

Second time denying, the app crashes
```
D/AndroidRuntime( 3792): Shutting down VM
E/AndroidRuntime( 3792): FATAL EXCEPTION: main
E/AndroidRuntime( 3792): Process: app.infmarketplace, PID: 3792
E/AndroidRuntime( 3792): java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=34, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {app.infmarketplace/app.infmarketplace.MainActivity}: java.lang.IllegalStateException: Reply already submitted
E/AndroidRuntime( 3792): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4324)
E/AndroidRuntime( 3792): 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4367)
E/AndroidRuntime( 3792): 	at android.app.ActivityThread.-wrap19(Unknown Source:0)
E/AndroidRuntime( 3792): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1649)
E/AndroidRuntime( 3792): 	at android.os.Handler.dispatchMessage(Handler.java:105)
E/AndroidRuntime( 3792): 	at android.os.Looper.loop(Looper.java:164)
E/AndroidRuntime( 3792): 	at android.app.ActivityThread.main(ActivityThread.java:6541)
E/AndroidRuntime( 3792): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 3792): 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
E/AndroidRuntime( 3792): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
E/AndroidRuntime( 3792): Caused by: java.lang.IllegalStateException: Reply already submitted
E/AndroidRuntime( 3792): 	at io.flutter.view.FlutterNativeView$1.reply(FlutterNativeView.java:162)
E/AndroidRuntime( 3792): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.error(MethodChannel.java:199)
E/AndroidRuntime( 3792): 	at com.baseflow.flutter.plugin.geolocator.data.Result.error(Result.java:30)
E/AndroidRuntime( 3792): 	at com.baseflow.flutter.plugin.geolocator.tasks.LastKnownLocationTask.handleError(LastKnownLocationTask.java:40)
E/AndroidRuntime( 3792): 	at com.baseflow.flutter.plugin.geolocator.tasks.LocationTask$1.onRequestPermissionsResult(LocationTask.java:118)
E/AndroidRuntime( 3792): 	at io.flutter.app.FlutterPluginRegistry.onRequestPermissionsResult(FlutterPluginRegistry.java:175)
E/AndroidRuntime( 3792): 	at io.flutter.app.FlutterActivityDelegate.onRequestPermissionsResult(FlutterActivityDelegate.java:132)
E/AndroidRuntime( 3792): 	at io.flutter.app.FlutterActivity.onRequestPermissionsResult(FlutterActivity.java:134)
E/AndroidRuntime( 3792): 	at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:7388)
E/AndroidRuntime( 3792): 	at android.app.Activity.dispatchActivityResult(Activity.java:7239)
E/AndroidRuntime( 3792): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4320)
E/AndroidRuntime( 3792): 	... 9 more
```

### Expected behavior

Return the exception again

### Reproduction steps

Request last known location as follows
```
 try {
    Position position = await Geolocator().getLastKnownPosition(LocationAccuracy.medium);
    print(position); // May be null
  } catch (ex) {
    print(ex); // Or fail to give permissions
    // PlatformException(PERMISSION_DENIED, Access to location data denied, null)
  }
```

### Configuration

**Version:** 1.4.0

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,24 @@
-## [1.4.0
+## [1.5.0]
+
+* It is now possible to check the location permissions using the `checkGeolocationStatus` method [[ISSUE #51](https://github.com/BaseflowIT/flutter-geolocator/issues/51)].
+* Improved the example App [[ISSUE #54](https://github.com/BaseflowIT/flutter-geolocator/issues/54)]
+* Solved a bug on Android causing a memory leak when you stop listening to the position stream.
+* **breaking** Solved a bug on Android where permissions could be requested more then once simultaniously [[ISSUE #58](https://github.com/BaseflowIT/flutter-geolocator/issues/58)]
+* Solved a bug on Android where requesting permissions twice would cause the App to crash [[ISSUE #61](https://github.com/BaseflowIT/flutter-geolocator/issues/61)]
+
+> **Important:**
+> 
+> To be able to correctly fix [issue #58](https://github.com/BaseflowIT/flutter-geolocator/issues/58) we had to change the `getPositionStream` method into a `async` method. This means the signature of the method has been changed from:
+>
+> `Stream<Position> getPositionStream([LocationOptions locationOptions = const LocationOptions()])` 
+>
+> to 
+>
+> `Future<Stream<Position>> getPositionStream([LocationOptions locationOptions = const LocationOptions()])`. 
+>
+> Meaning as a developer you'll now have to `await` the result of the method to get access to the actual stream.
+
+## [1.4.0]
 
 * Added feature to query the last known location that is stored on the device using the `getLastKnownLocation` method;
 * **breaking** Renamed the `getPosition` to `getCurrentPosition`;

--- a/README.md
+++ b/README.md
@@ -24,7 +24,7 @@ To use this plugin, add `geolocator` as a [dependency in your pubspec.yaml file]
 
 ```yaml
 dependencies:
-  geolocator: '^1.4.0'
+  geolocator: '^1.5.0'
 ```
 
 > **NOTE:** There's a known issue with integrating plugins that use Swift into a Flutter project created with the Objective-C template. See issue [Flutter#16049](https://github.com/flutter/flutter/issues/16049) for help on integration.

--- a/ios/geolocator.podspec
+++ b/ios/geolocator.podspec
@@ -3,7 +3,7 @@
 #
 Pod::Spec.new do |s|
   s.name             = 'geolocator'
-  s.version          = '1.4.0'
+  s.version          = '1.5.0'
   s.summary          = 'Geolocation plugin for Flutter. This plugin provides a cross-platform API for generic location (GPS etc.) functions.'
   s.description      = <<-DESC
 Geolocation plugin for Flutter.

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -1,6 +1,6 @@
 name: geolocator
 description: Geolocation plugin for Flutter. This plugin provides a cross-platform (iOS, Android) API for generic location (GPS etc.) functions.
-version: 1.4.0
+version: 1.5.0
 author: Baseflow <hello@baseflow.com>
 homepage: https://github.com/baseflowit/flutter-geolocator
 

